module/version_reconfigure_test.go

```diff
@@ -0,0 +1,235 @@
+	test.That(t, err, test.ShouldBeNil)
+
+	serverPath, err := testutils.BuildTempModule(t, "web/cmd/server/")
+	test.That(t, err, test.ShouldBeNil)
+
+	server := pexec.NewManagedProcess(pexec.ProcessConfig{
+		Name: serverPath,
+		Args: []string{"-config", cfgFilename},
+		CWD:  utils.ResolveFile("./"),
+		Log:  true,
+	}, logger)
+
+	err = server.Start(context.Background())
+	test.That(t, err, test.ShouldBeNil)
+	defer func() {
+		test.That(t, server.Stop(), test.ShouldBeNil)
+	}()
+
+	rc, err := connect(port, logger)
+	test.That(t, err, test.ShouldBeNil)
+	defer func() {
+		test.That(t, rc.Close(context.Background()), test.ShouldBeNil)
+	}()
+
+	// Assert that motors and base were added.
+	_, err = rc.ResourceByName(generic.Named("generic1"))
+	test.That(t, err, test.ShouldBeNil)
+
+	// Assert that there were no validation or component building errors
+	test.That(t, logs.FilterMessageSnippet(
+		"modular config validation error found in resource: generic1").Len(), test.ShouldEqual, 0)
+	test.That(t, logs.FilterMessageSnippet("error building component").Len(), test.ShouldEqual, 0)
+
+	// Read the config, swap to `run_version2.sh`, and overwrite the config, triggering a reconfigure where `generic1` will fail validation
+	cfg, err := config.Read(context.Background(), cfgFilename, logger)
+	test.That(t, err, test.ShouldBeNil)
+	cfg.Modules[0].ExePath = utils.ResolveFile("module/multiversionmodule/run_version2.sh")
+	newCfgBytes, err := json.Marshal(cfg)
+	test.That(t, err, test.ShouldBeNil)
+	os.WriteFile(cfgFilename, newCfgBytes, 0o644)
+
+	// Wait for reconfiguration to finish with a 30 second timeout.
+	reconfigureCheckTicker := time.NewTicker(time.Second)
+	reconfigureFinished := false
+	reconfigureCtx, reconfigureCancel := context.WithTimeout(context.Background(), 30*time.Second)
+	defer reconfigureCancel()
+	for !reconfigureFinished {
+		select {
+		case <-reconfigureCheckTicker.C:
+			_, err = rc.ResourceByName(generic.Named("generic1"))
+			if err != nil {
+				reconfigureFinished = true
+			}
+		case <-reconfigureCtx.Done():
+			test.That(t, reconfigureCtx.Err(), test.ShouldBeNil)
+		}
+	}
+
+	// Check that the motors are still present but that "base1" is not started
+	_, err = rc.ResourceByName(generic.Named("generic1"))
+	test.That(t, err, test.ShouldNotBeNil)
+	test.That(t, err.Error(), test.ShouldResemble, `resource "rdk:component:generic/generic1" not found`)
+
+	// Assert that Validation failure message is present
+	//
+	// Race condition safety: Resource removal should occur after modular resource validation (during completeConfig), so if
+	// ResourceByName is failing, these errors should already be present
+	test.That(t, logs.FilterMessageSnippet(
+		"modular config validation error found in resource: generic1").Len(), test.ShouldEqual, 1)
+	test.That(t, logs.FilterMessageSnippet("error building component").Len(), test.ShouldEqual, 0)
+}
+
+func TestVersionBumpWithNewImplicitDeps(t *testing.T) {
+	t.Skip()
```

Comment:
@dgottlieb  This test still fails. :( I only realized that this error case exists after fixing the first one.